src/main/java/net/dries007/tfc/util/InteractionManager.java

```diff
@@ -305,6 +309,144 @@ else if (!level.getBlockState(relativePos.below()).isAir())
+                            {
+                                // Shrinking is already handled by the placement onItemUse() call, we just need to insert the stack
+                                stackBefore.setCount(1);
+                                level.getBlockEntity(topPos, TFCBlockEntities.INGOT_PILE.get()).ifPresent(topPile -> topPile.addIngot(stackBefore));
+                            }
+                            return InteractionResult.SUCCESS;
+                        }
+                        return InteractionResult.FAIL;
+                    }
+                }
+                else
+                {
+                    // We clicked on a non-ingot pile, so we want to try and place an ingot pile at the current location.
+                    final ItemStack stackBefore = stack.copy();
+                    final InteractionResult result = ingotPilePlacement.onItemUse(stack, context);
+                    if (result.consumesAction())
+                    {
+                        // Shrinking is already handled by the placement onItemUse() call, we just need to insert the stack
+                        stackBefore.setCount(1);
+                        level.getBlockEntity(relativePos, TFCBlockEntities.INGOT_PILE.get()).ifPresent(pile -> pile.addIngot(stackBefore));
+                    }
+                    return result;
+                }
+            }
+            return InteractionResult.PASS;
+        });
+
+        final BlockItemPlacement sheetPilePlacement = new BlockItemPlacement(() -> Items.AIR, TFCBlocks.SHEET_PILE);
+        register(TFCTags.Items.PILEABLE_SHEETS, false, (stack, context) -> {
+            final Player player = context.getPlayer();
+            if (player != null && player.isShiftKeyDown())
+            {
+                final Level level = context.getLevel();
+                final Direction clickedFace = context.getClickedFace(); // i.e. click on UP
+                final Direction sheetFace = clickedFace.getOpposite(); // i.e. place on DOWN
+                final BlockPos relativePos = context.getClickedPos().relative(clickedFace);
+                final BlockState relativeState = level.getBlockState(relativePos);
+
+                final BlockPlaceContext blockContext = new BlockPlaceContext(context);
+                final BooleanProperty property = DirectionPropertyBlock.getProperty(sheetFace);
+
+                if (blockContext.replacingClickedOnBlock())
+                {
+                    // todo: this causes weird behavior, how to handle properly?
+                    return InteractionResult.FAIL;
+                }
+
+                // Sheets behave differently than ingots, because we need to check the targeted face if it's empty or not
+                // We assume immediately that we want to target the relative pos and state
+                if (Helpers.isBlock(relativeState, TFCBlocks.SHEET_PILE.get()))
+                {
```

Comment:
IHighlightHandler could help with players targeting

Reply:
While true, this isn't the case where they're needed (since this is just placing a block targeting an unrelated one), it's already pretty visible when breaking, and I'm lazy so I think I'll throw a todo and leave it at that